File: WebApplication/src/voucher-requests/dtos/voucher-request.dto.ts
import { CompanyDto } from "src/companies/dtos/company.dto";
import { SupplierOfferingDto } from "src/suppliers/dtos/supplier-offering.dto";
import { VoucherRequestState } from "../enums/voucher-request-state.enum";

export class VoucherRequestDto {
    id: number;

    code: string;

    state: VoucherRequestState;

    createDate: Date;

    requestUserId: number;
    requestCompanyId: number;
    requestCompany: CompanyDto;

    supplierOfferingId: number;
    supplierOffering: SupplierOfferingDto;
    declineReason: string;
}

File: Sc.Models/Dtos/VoucherRequests/VoucherRequestDto.cs
using Infrastructure.DomainValidation;
using Infrastructure.DomainValidation.Models.ErrorCodes;
using Sc.Models.Dtos.Companies;
using Sc.Models.Dtos.Suppliers;
using Sc.Models.Entities.Base;
using Sc.Models.Enums.VoucherRequests;
using Sc.Models.Interfaces;

namespace Sc.Models.Dtos.VoucherRequests
{
    public class VoucherRequestDto : Entity, IValidate
    {
        public string Code { get; set; }

        public VoucherRequestState State { get; set; }

        public DateTime CreateDate { get; set; }

        public int RequestUserId { get; set; }
        public int RequestCompanyId { get; set; }
        public CompanyDto RequestCompany { get; set; }

        public int SupplierOfferingId { get; set; }
        public SupplierOfferingDto SupplierOffering { get; set; }
        public string DeclineReason { get; set; }

        public void ValidateProperties(DomainValidatorService domainValidatorService)
        {
            if (RequestUserId == 0)
            {
                domainValidatorService.ThrowErrorMessage(VoucherRequestErrorCode.VoucherRequest_User_Required);
            }

            if (RequestCompanyId == 0)
            {
                domainValidatorService.ThrowErrorMessage(VoucherRequestErrorCode.VoucherRequest_Company_Required);
            }

            if (SupplierOfferingId == 0)
            {
                domainValidatorService.ThrowErrorMessage(VoucherRequestErrorCode.VoucherRequest_SupplierOffering_Required);
            }
        }
    }
}
